### ISSUE TITLE
(Re-add) Only compile Sass files that have been modified

### ISSUE THREAD

**Issue by addyosmani** [google/web-starter-kit#472]
In 0.5.x, we dropped support for only recompiling Sass files that have changed. This is pretty painful during `gulp serve` in the material-sprint branch :)

This feature was reverted because users were running into issues with it not correctly reloading on save.

Let's try adding back in support. We'll probably want to look at updating paths and any open issues around browser-sync in this area.

**Comment by addyosmani** [comment]
cc @sindresorhus who wrote gulp-changed and IIRC, added in our support for it for thoughts on the above comments. Thanks @DominicBarnard.

**Comment by adyz** [comment]
I have this issue with BrowserSync & Gulp alone following BrowserSync's guide for Gulp.
http://www.browsersync.io/docs/gulp/


If I have two files with .scss extention and I change either one, it will recompile both:

```
[BS] 2 files changed (main.css, secondary.css)
```

The same thing happens with Web-Starter-Kit.

**Comment by sindresorhus** [comment]
Yes, gulp-changed only handles 1:1 relationships. Maybe try using gulp-cached + gulp-remember instead.

**Comment by cmndo** [comment]
I found this thread after attempting to solve this same problem on my own. I'm no where near as experienced with the codebase as @sindresorhus.

I've found [https://github.com/xzyfer/sass-graph] which builds a dependency chart for @imports.This seemed like a good method to explore. before i start watching for file changes I save the graph into a variable
    
    var graph = sassGraph.parseDir(config.paths.src + '/scss/pages');
    gulp.watch('public_html/src/scss/**/*.scss').on('change', function(file){
        var imports = config.graph.index[file.path].importedBy;
        return sass(imports).pipe(gulp.dest('public_html/dist/css'));
    });

Of course that doesn't work, but the idea might. That would at least cut down on the number of files sass is building on each change.

**Comment by sindresorhus** [comment]
@cmndo Yup, having the dependency graph is the only way to reliably cache the Sass compilation.

**Comment by robwierzbowski** [comment]
Because:
- libsass maintainers suggest that libsass is fast enough to not worry about caching. 
- Because Sass has its own import system, comparing file timestamps doesn't work. As mentioned above, if you really want to get incremental you need a dependency tree.

Would we gain enough speed to make adding a couple more plugins and complexity to the styles task worth it? My styles refresh pretty damn fast (on a MacBook Pro 2015 13"). 

Based on my experience I would be happy only filtering `changed` files for CSS.
